### ISSUE TITLE
Ambigous query produced on upsert attempt

### ISSUE THREAD

**Issue by aerosol** [elixir-ecto/ecto#1914]
### Environment

* Elixir version (elixir -v): 1.3.2
* Database and version: PostgreSQL 9.6.1
* Ecto version (mix deps): 2.1.2
* Database adapter and version (mix deps):
  - postgrex 0.13.0 (Hex package) (mix)
  locked at 0.13.0 (postgrex) e101ab47

* Operating system: Darwin Kernel Version 16.3.0

### Current behavior

Trying to perform upsert with counter increase on conflict (unique constraint violation).

Migration:

```elixir
create table(:people) do
   add :email, :string, null: false
   add :times_seen, :integer, default: 1
   timestamps
end

create unique_index(:people, [:email])
```

Upsert:

```elixir
l = [%{email: "foo@example.com", updated_at: now, inserted_at: now}]
```

```elixir
"people"
|> Repo.insert_all(l, 
  returning: [:id, :email], 
  on_conflict: [
    inc: [times_seen: 1]
  ], conflict_target: :email)
```

Produces an error:

```
** (Postgrex.Error) ERROR 42702 (ambiguous_column): column reference "times_seen" is ambiguous
```

Query:

```sql
INSERT INTO "people" AS s0 ("email","inserted_at","updated_at") 
VALUES ($1,$2,$3) ON CONFLICT ("email") 
DO UPDATE SET "times_seen" = "times_seen" + $4 
RETURNING "id", "email"
```

Bindings:

```elixir
 ["foo@example.com", "2017-01-17 10:57:51", "2017-01-17 10:57:51", 1]
```

The problem with generated query seems to be:

```sql
SET "times_seen" = "times_seen" + $4
```

### Expected behavior

```sql
SET "times_seen" = "s0"."times_seen" + $4
```

**Comment by aerosol** [comment]
Failing test case:

```elixir
    @tag :with_conflict_target
    test "on conflict with inc" do
      uuid = "6fa459ea-ee8a-3ca4-894e-db77e160355e"
      post = %Post{title: "first", uuid: uuid}
      {:ok, _} = TestRepo.insert(post)

      post = %{title: "upsert", uuid: uuid}

      TestRepo.insert_all(Post, [post], on_conflict: [inc: [visits: 1]], conflict_target: :uuid)
    end
```

**Comment by josevalim** [comment]
Beautiful! I will work on a fix today and likely get a new v2.1 out as we have had other bug fixes lately.

**Comment by michalmuskala** [comment]
I think the final query should rather be:
```sql
SET "s0"."times_seen" = "s0"."times_seen" + $4
```
so we include the table reference in all the places.

**Comment by josevalim** [comment]
@michalmuskala just FYI, that query is not supported. You can't have "s0" on the left side of `=`.

**Comment by josevalim** [comment]
Fixed on e25aed8. v2.1.2 coming out shortly. Thank you for reporting and the test case @aerosol!

**Comment by aerosol** [comment]
@josevalim thank you, really appreciate your work!
